src/main/java/com/dev/cinema/service/impl/AuthenticationServiceImpl.java

```diff
@@ -28,6 +32,7 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
+        shoppingCartService.registerNewShCart(user);
         return userService.add(user);
```

Comment:
Shouldn't we create the user before creating a shopping cart for him?

Reply:
agree